src/main/webapp/js/services.js

```diff
@@ -2,16 +2,20 @@
 'use strict';
 
 angular.module('app.services')
-.service('IdeasSelectedByTag',  ['Data', IdeasSelectedByTag]); 
-function IdeasSelectedByTag (Data) { 
+.service('IdeasSelectedByTag',  ['ideasFactory', IdeasSelectedByTag]);
+function IdeasSelectedByTag (ideasFactory) {
        return {
         getIdeas:function (tag) {
-          var newData = Data.getIdeas();
+         var newData;
+         ideasFactory.getIdeas().then(function (ideas) {
+            newData = ideas;
+         });
+
           if(tag !== undefined) {
               newData = [];
```

Comment:
This method should be refactored to call factory method only once to to decrease number of server calls.